OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -136,7 +134,7 @@ public class Armament : PausableConditionalTrait<ArmamentInfo>, ITick
 		public Armament(Actor self, ArmamentInfo info)
 			: base(info)
 		{
-			this.self = self;
+			Actor = self;
```

Comment:
In the future we may want to consider refactoring `AttackGarrisoned` to remove this